### ISSUE TITLE
Feature Request: Support loading multi-disc games with m3u files

### ISSUE THREAD

**Issue by orbea** [flyinghead/flycast#423]
**Is your feature request related to a problem? Please describe.**
Can `.m3u` files be supported for multi-disc games?

**Describe the solution you'd like**
In other emulators (mednafen) for consoles that have multi-disc games the standard way of loading these games are plain text `.m3u` files. While perhaps it should not be required, it would be nice if it worked with flycast too.

For example in mednafen.

> $ ls
Koudelka (USA) (Disc 1).bin  Koudelka (USA) (Disc 2).bin  Koudelka (USA) (Disc 3).bin  Koudelka (USA) (Disc 4).bin  Koudelka (USA).m3u
Koudelka (USA) (Disc 1).cue  Koudelka (USA) (Disc 2).cue  Koudelka (USA) (Disc 3).cue  Koudelka (USA) (Disc 4).cue
>
> $ cat Koudelka\ \(USA\).m3u 
./Koudelka (USA) (Disc 1).cue
./Koudelka (USA) (Disc 2).cue
./Koudelka (USA) (Disc 3).cue
./Koudelka (USA) (Disc 4).cue

And with flycast it could be useful in a case like this.

> $ ls
Skies of Arcadia (USA) (Disc 1).cue            Skies of Arcadia (USA) (Disc 1) (Track 3).bin  Skies of Arcadia (USA) (Disc 2) (Track 2).bin
Skies of Arcadia (USA) (Disc 1) (Track 1).bin  Skies of Arcadia (USA) (Disc 2).cue            Skies of Arcadia (USA) (Disc 2) (Track 3).bin
Skies of Arcadia (USA) (Disc 1) (Track 2).bin  Skies of Arcadia (USA) (Disc 2) (Track 1).bin  Skies of Arcadia (USA).m3u
>
> $ cat Skies\ of\ Arcadia\ \(USA\).m3u 
./Skies of Arcadia (USA) (Disc 1).cue
./Skies of Arcadia (USA) (Disc 2).cue

And then I would be able to start the game.
```
flycast 'Skies of Arcadia (USA).m3u'
```

**Describe alternatives you've considered**
What is the intended way of loading multi-disc games with flycast? I haven't found much documentation, but maybe I am just looking in the wrong place.

**Comment by adamsbuilds** [comment]
I think this is the single most important thing that is missing from flycast.   Is it considered to be added?   If not liberto core is my only option

**Comment by cinnamonmatexfce** [comment]
> I think this is the single most important thing that is missing from flycast.   Is it considered to be added?   If not liberto core is my only option

You know you can swap discs from the Flycast quick menu (open it pressing >TAB< - by default - on the keyboard)?

**Comment by flyinghead** [comment]
This feature request has low priority because it's not blocking. You can eject the current disk and select a different one as cinnamonmatexfce said. Supporting .m3u would require additional UI to be able to switch to the next/previous disk in the list.

**Comment by orbea** [comment]
> Supporting .m3u would require additional UI to be able to switch to the next/previous disk in the list.

At minimum would be two keybinds, the first to eject/insert and the second to select the next disc.

**Comment by adamsbuilds** [comment]
> 

Yes but this isn't really practical in situations not involving a mouse and keyboard.   For example I use batocera.linux where everything is with a controller.  And most the other disc based emulators have adopted this.. dolphin even has a method of automatically switching discs while others simply have some type of key combination required.  I would recommend dolphins approach because it's pretty dang amazing.   So for me I side batocera the liberto core is my best option as RetroArch can control the disc swap method pretty well.  In any case would love to see the similar added to the standalone (.m3u support, preferably auto disc swap like dolphin)

**Comment by flyinghead** [comment]
You can switch disc with a controller. You don't need a mouse or keyboard.

**Comment by adamsbuilds** [comment]
Elaborate please.  If it can't load an m3u files how can it swap
"tracks/disc" without loading an entirely new rom manually.  In my case
loading from emulation station doesn't seem very practical.  Wouldn't an
m3u method of loading all discs be better ui?

**Comment by flyinghead** [comment]
When you want to switch disc, pause the game with the Menu button and select Eject Disk:
![image](https://user-images.githubusercontent.com/16261910/192154228-70a9ed55-9d43-46eb-a20e-b684afa8e026.png)
Then pause the game again with the same button and select Insert Disk:
![image](https://user-images.githubusercontent.com/16261910/192154266-3a65f36e-3f20-4c74-8b3b-65bbf1fc2cc3.png)

**Comment by adamsbuilds** [comment]
Thank you for the response I will mess around with it and see if I can find
a method to work for my application.  Or maybe just use the liberto core
for multi disc games only and use the standalone for everything else .
This is what I currently do out of emulation station.   Thanks for the
great work on this emulator to anyone involved.  Do you forsee further
upgraded support of multidisc based games in flycast in the future?

**Comment by roguesaloon** [comment]
On a steam deck here (using the flatpak version) and I would also really love to see native M3U support. The issue for me isn't about not being able to switch discs when required, but about selecting the discs with a single file/shortcut per-game. 

Steam Rom Manager gets confused if you have duplicates, so you put the multi-discs (chd's/cdi's) in a sub-folder to hide them, you also want to keep the game in the roms folder which is where an M3U comes in (To tell it where to find both discs). Right now, this partially works for SRM with the caveat that flycast doesn't know what to do with the M3U, so you have to edit the steam shortcut manually to point to disc 1 and change this again to disc 2 manually when required (SRM at least doesn't alter this on a re-import, so this is a hacky workaround).

If Flycast had an in-emulator concept of a multiple disc games from a single file (typically an M3U file), and let you launch the last opened disc for a game from there, it eliminates the issue entirely. Otherwise, you would require either two entries in your steam library or manually editing the game shortcut (as described above).

**Comment by SixWingedFreak** [comment]
+1 For m3u support. I use .m3u files in LaunchBox for multi-disc games so that I don't have two to four instances of the same game. Looks a lot prettier. Adjust the UI so it has Eject Disk/Insert Disc on the same button as you have it, and add Previous Disk and Next Disk, but are greyed out when it loads a file that's not m3u.

**Comment by KoB-Kirito** [comment]
+1
I'm on JELOS (RGB30) and flycast just exits when you eject a disc.

**Comment by flyinghead** [comment]
You need to add content folders so that Flycast can find your disks.

**Comment by byjove01** [comment]
I just want to bring my support to that feature. It would be really something to be able to load multi-disc games without using the UI to eject/re-insert disk images.

**Comment by Hunk4TH** [comment]
+1 for M3u support!

**Comment by cornfeedhobo** [comment]
Stuck on this for handhelds that use retroarch. I don't see where one can configure Flycast to point at a directory of discs. Retroarch is able to load the M3U and pops up with a notification that it knows it's loading multiple discs, but no clue how to load the second disc.

Anyone have any ideas?

**Comment by SpikeTheDragon40k** [comment]
+1 for M3u support!

**Comment by MrGrosNounours** [comment]
I need this feature too, so +1!

**Comment by discus7l** [comment]
+1 for M3u support!
